pkg/sink/initialization_test.go

```diff
@@ -101,3 +111,54 @@ func Test_GetArgs_error(t *testing.T) {
 		})
 	}
 }
+
+func TestWaitForEventlistener(t *testing.T) {
+	eventListenerName := "my-eventlistener"
+	namespace := "my-namespace"
+	eventListener := &triggersv1.EventListener{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      eventListenerName,
+			Namespace: namespace,
+		},
+		Spec: triggersv1.EventListenerSpec{},
+	}
+	ctx, _ := rtesting.SetupFakeContext(t)
+	test.SeedResources(t, ctx, test.Resources{EventListeners: []*triggersv1.EventListener{eventListener}})
+	r := Sink{
+		EventListenerName:      eventListenerName,
+		EventListenerNamespace: namespace,
+		EventListenerLister:    eventlistenerinformer.Get(ctx).Lister(),
+	}
+
+	r.WaitForEventListenerOrDie()
+}
+
+func TestWaitForEventlistener_Fatal(t *testing.T) {
+	if os.Getenv("EL_FATAL_CRASH") == "1" {
+		eventListenerName := "my-eventlistener"
+		namespace := "my-namespace"
+		ctx, _ := rtesting.SetupFakeContext(t)
+		test.SeedResources(t, ctx, test.Resources{EventListeners: []*triggersv1.EventListener{}})
+		r := Sink{
+			EventListenerName:      eventListenerName,
+			EventListenerNamespace: namespace,
+			EventListenerLister:    eventlistenerinformer.Get(ctx).Lister(),
+		}
+		elBackoff = wait.Backoff{
+			Duration: 50 * time.Millisecond,
+			Factor:   1.0,
+			Jitter:   0.1,
+			Steps:    1,
+			Cap:      100 * time.Millisecond,
+		}
+		// will fail
+		r.WaitForEventListenerOrDie()
+	}
+	cmd := exec.Command(os.Args[0], "-test.run=TestWaitForEventlistener_Fatal") //nolint:gosec
+	cmd.Env = append(os.Environ(), "EL_FATAL_CRASH=1")
```

Comment:
this is clever...though we might be able to get away with passing in a test logger and capturing its output and then comparing (like we do here: https://github.com/tektoncd/triggers/blob/1066d18ecb7b1e4ca88472251a2561e5c684046c/pkg/sink/sink_test.go#L926-L943)

What do you think?

Another option would be to have the function return an error and then have main.go call the log.Fatal.

Reply:
yes, let me take a look at this strategy and try to implement it here

Reply:
so, it doesn't look like this is possible because the behavior of fatal can't be overridden even though there is a noop action in the library. basically, its not a supported action on fatal.

i updated this function to return an error and panic instead